picky/src/x509/wincert.rs

```diff
@@ -261,21 +76,18 @@ impl WinCertificate {
             certificate,
         } = self;
 
-        let mut buffer = BufWriter::new(Vec::new());
-
         let padding = (8 - (certificate.len() % 8)) % 8;
 
+        let mut buffer = BufWriter::with_capacity(length as usize + padding, Vec::new());
+
+        // According to Authenticode_PE.docx WinCertificate::length is set to certificate length,
+        // but if do not contains others WinCertificate's fields sizes the signature is invalid
```

Comment:
I don’t understand this part of the comment (I’ll check Authenticode_PE.docx)

Reply:
Okay, I read about `_WIN_CERTIFICATE`, what did you mean with the second part of the comment?

Reply:
Authenticode_PE.docx says that `WinCertificate::length` is the length of  `WinCertificate::certificate`, but `WinCertificate::length` should include sizes of other WinCertificate's fields(`WinCertifcate::revision`, `WinCertificate::certificate_type` and `WinCertifcate::length` itself). Otherwise the singature will be invalid. MSDN says the same **dwLength** - `Specifies the length, in bytes, of the signature.` I should rephrase the comment and put it near `WinCertificate::from_certificate`

Reply:
Oh I understand. Sure, do that :slightly_smiling_face: